nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_constructor.rs

```diff
@@ -112,12 +116,35 @@ impl StringConstructor {
         Ok(s.into_value())
     }
 
+    /// ### [22.1.2.1 String.fromCharCode ( ...`codeUnits` )](https://262.ecma-international.org/15.0/index.html#sec-string.fromcharcode)
+    ///
+    /// > This function may be called with any number of arguments which form
+    /// the rest parameter `codeUnits`.
     fn from_char_code(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _this_value: Value,
-        _arguments: ArgumentsList,
+        code_units: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!();
+        // 1. Let result be the empty String.
+        if code_units.is_empty() {
+            return Ok(String::EMPTY_STRING.into_value());
+        }
+        let mut result: Vec<u8> = Vec::with_capacity(code_units.len() * 2);
+
+        // 2. For each element next of codeUnits, do
+        //   a. Let nextCU be the code unit whose numeric value is ℝ(? ToUint16(next)).
+        //   b. Set result to the string-concatenation of result and nextCU.
+        for next in code_units.iter() {
+            let code_unit = next.to_uint16(agent)?;
```

Comment:
**issue:** You've effectively only performed the `R(? ToUint16(next))` part but are missing the `be the code unit` step. This is referring to the Unicode code units, so effectively a piece of text. This is probably the API you really want to be using here: https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf16_lossy

Reply:
I spent an hour looking for that API... thank you!